Bump Tested Up To Versions for 3.6 Release

jessy-p opened the pull request:
Bumps the tested to version based on any available release candidates for WordPress or WooCommerce in this or the next WCPay release cycle. Updates the minimum required version by  increasing the version by 1.

#### Changes proposed in this Pull Request
- Bumped the tested to version based on:
   - [WordPress 5.9 release date is January 25, 2022](https://make.wordpress.org/core/5-9/)
   - [WooCommerce 6.1 release date is January 11, 2022](https://developer.woocommerce.com/2021/12/28/woocommerce-6-1-rc1/)
- Updated the minimum required version
  - WordPress 5.6 to 5.7
  - WooCommerce 4.4 to 4.5
- Updated CI test versions

#### Testing instructions
- Validate the changes in `woocommerce-payments.php` and `readme.txt` 
- CI tests have been updated, should pass and cover all required compatabilities.

-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

htdat reviewed (changes requested):
> Validate the changes in woocommerce-payments.php and readme.txt for Tested up to

✅ This part looks good to me!

> Validate that no changes are required for **_Requires at least_**, as per the [L-2 transition timeline](https://developer.woocommerce.com/2021/05/12/woocommerce-payments-is-adopting-a-new-version-support-policy/).

I think this doc is obsolete. We need to follow this part in PaJDYF-1iC-p2 especially the underline texts:

![Markup on 2022-01-04 at 11:29:17](https://user-images.githubusercontent.com/10045087/148009639-eaa7824e-5fbb-4e4b-af34-6de9277f2c31.png)

So I think we will need to change this to 4.5 and update CI tests.

https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/woocommerce-payments.php#L11

htdat commented:
> So I think we will need to change this to 4.5 and update CI tests.

I've spotted a few CI tests to update: 

- Change 4.4.0 to 4.5.0:
https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/.github/workflows/e2e-test.yml#L38-L39

- Remove 4.4.0 for matrix. For the edge case, change 4.4.0 to 4.5.0 - may we need to update WP version from 5.4 to 5.5 too? 
https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/.github/workflows/compatibility.yml#L13-L21

--- 

Also, this would change to 5.6.0 as the current L-2 of Woo core, which would have been done in the previous WCPay release.

https://github.com/Automattic/woocommerce-payments/blob/f582049892d773aeb781b311447f5cf207af514c/.github/workflows/php-lint-test.yml#L8-L9

jrodger reviewed (dismissed):
Thanks for tackling this new process @jessy-p. The changes look good to me and I agree with the points that @htdat raised. I think we're good to merge this once those are addressed.

jessy-p commented:
Addressed all review comments except  update wordpress from 5.4 to 5.6  in oldest dependency compatibility because of [ an issue](p1641279509415000/1641232729.391900-slack-CGGCLBN58)

htdat commented on phpcs-compat.xml.dist:
```diff
@@ -15,7 +15,7 @@
 
 	<!-- Configs -->
-	<config name="minimum_supported_wp_version" value="5.6" />
+	<config name="minimum_supported_wp_version" value="5.7" />
```
WCPay 3.6 is released on Jan 19 while WP core 5.9 is released later on Jan 25. I think it would be better to update this minimum supported version for WP (here and other places) in WCPay 3.7?

jessy-p replied:
Fixed

htdat commented on .github/workflows/compatibility.yml:
```diff
@@ -11,12 +11,12 @@ jobs:
           # Edge case: oldest dependencies compatibility
-          - woocommerce: '4.4.0'
+          - woocommerce: '4.5.0'
             wordpress:   '5.4'
```
I think it's still good to update this WP version from 5.4 to 5.6. That's because, in practice, WCPay is requiring WP version >= 5.6.

jessy-p replied:
But the E2E tests are failing. I think we need to fix that first then.

htdat replied:
> I think we need to fix that first then ( on a separate PR ).

👍 Agreed!

htdat replied:
Let me spin up another PR as I had some ideas about it in this test PR https://github.com/Automattic/woocommerce-payments/pull/3346

htdat replied:
> Let me spin up another PR as I had some ideas about it in this test PR #3346

Here is the PR https://github.com/Automattic/woocommerce-payments/pull/3597

jessy-p commented:
>Addressed all review comments except update wordpress from 5.4 to 5.6 in oldest dependency compatibility 
Fixed this as well.

htdat commented on .github/workflows/php-lint-test.yml:
```diff
@@ -4,8 +4,8 @@ on:
   pull_request
 
 env:
+  WC_VERSION: 5.6.0  # the min supported version as per L-2 policy
```
Nitpick: may only change the version number rather than change its line?

jessy-p replied:
Fixed

htdat commented on readme.txt:
```diff
@@ -1,8 +1,8 @@
 Tags: woocommerce, payment, payment request, credit card, automattic
-Requires at least: 5.6
-Tested up to: 5.8.2
+Requires at least: 5.7
```
May revert this change back to 5.6?

htdat commented on changelog.txt:
```diff
@@ -1,6 +1,8 @@
 
 = 3.6.0 - 2022-xx-xx =
+* Update - Bump minimum required version of WooCommerce from 4.4 to 4.5.
+* Update - Bump minimum required version of WordPress from 5.6 to 5.7.
```
May remove this line?

jessy-p replied:
Removed

htdat commented on readme.txt:
```diff
@@ -38,7 +38,7 @@ Our global support team is available to answer questions you may have about WooC
 = Requirements =
 
-* WordPress 5.6 or newer.
+* WordPress 5.7 or newer.
```
Revert this change too?

htdat commented on readme.txt:
```diff
@@ -99,6 +99,8 @@ Please note that our support for the checkout block is still experimental and th
 
 = 3.6.0 - 2022-xx-xx =
+* Update - Bump minimum required version of WooCommerce from 4.4 to 4.5.
+* Update - Bump minimum required version of WordPress from 5.6 to 5.7.
```
May remove this line too?

htdat commented on woocommerce-payments.php:
```diff
@@ -8,9 +8,9 @@
- * Requires at least: 5.6
+ * WC requires at least: 4.5
+ * WC tested up to: 6.1
+ * Requires at least: 5.7
```
May revert this change back to 5.6 too?

htdat reviewed (commented):
LGTM! After reverting back changes related to WP minimum version, then it's ready to 🚢
